### PR TITLE
[5.0] AVIF support for media manager

### PR DESCRIPTION
**Pull request by wilsonge**
Allow media manager to have AVIF files uploaded and rendered

### Summary of Changes
AVIF is an alternative to webp in some circumstances (see https://www.smashingmagazine.com/2021/09/modern-image-formats-avif-webp/ for more info).

I've deliberately chosen to not add this to a SQL Update script so there are no security implications for sites who may have more restrictive upload policies. This will only apply to new sites.

Worth noting this is a better fit for J5 than J4.4 because the GD library support was only added in PHP 8.1 (although technically that doesn't stop people uploading the files themselves in J4.4 as web browser support already exists - it's just easier because they can be edited too) - https://php.watch/versions/8.1/gd-avif

### Testing Instructions
Upload AVIF file. Before will be rejected and any files on the server won't be rendered. Afterwards they will be.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@wilsonge You've changed the base.sql file for MySQL but not the one for PostgreSQL. This cannot be right.

**Comment by wilsonge** [comment]
https://github.com/joomla/joomla-cms/pull/35752 is just for the image list field (I'm not aware of anywhere we use that in core). But people would have needed to upload those files by hand. This PR specifically addresses media manager.

Yes it will address https://github.com/joomla/joomla-cms/issues/30666. However it will only address the AVIF part of https://github.com/joomla/joomla-cms/issues/37669 (the original issue was a broader thing around document formats etc which obviously isn't covered here)

**Comment by Quy** [comment]
Update?

https://github.com/joomla/joomla-cms/blob/40bfac918d3f2de688d91ef8c34e0f1a5014710a/libraries/src/Helper/MediaHelper.php#L57

https://github.com/joomla/joomla-cms/blob/40bfac918d3f2de688d91ef8c34e0f1a5014710a/libraries/src/Helper/MediaHelper.php#L144

https://github.com/joomla/joomla-cms/blob/40bfac918d3f2de688d91ef8c34e0f1a5014710a/libraries/src/Helper/MediaHelper.php#L253

https://github.com/joomla/joomla-cms/blob/40bfac918d3f2de688d91ef8c34e0f1a5014710a/libraries/src/Helper/MediaHelper.php#L271

File: administrator/components/com_media/config.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
 			type="textarea"
 			label="COM_MEDIA_FIELD_RESTRICT_UPLOADS_EXTENSIONS_LABEL"
 			description="COM_MEDIA_FIELD_RESTRICT_UPLOADS_EXTENSIONS_DESC"
-			default="bmp,gif,jpg,jpeg,png,webp,ico,mp3,mp4,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,csv"
+			default="bmp,gif,jpg,jpeg,png,webp,avif,ico,mp3,mp4,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,csv"
```
Comment (Quy) [inline administrator/components/com_media/config.xml:58]: Entries are in alpha order except for `csv`.  Probably should do the same with `avif`.

**Comment by tomsrocket** [comment]
I can confirm this works.

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on d0ec8004db68c1cda2fcdf219707248584b75987<br /><br />On 5.0.0-alpha4
you have to setup the file type and mime type in options

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41381">issues.joomla.org/tracker/joomla-cms/41381</a>.</sub>

**Comment by Magnytu2** [comment]
It works but doesn't produce a thumbnail in the media manager, and the instructions for adding .avif files are missing from the settings.screen shot 2023-08-26 at 13 34 54<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41381">issues.joomla.org/tracker/joomla-cms/41381</a>.</sub>

File: administrator/components/com_media/config.xml (hunk 1 above)
Comment (Quy) [inline administrator/components/com_media/config.xml:58]:
```suggestion
			default="avif,bmp,gif,jpg,jpeg,png,webp,ico,mp3,mp4,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,csv"
```

File: administrator/components/com_media/config.xml [hunk 2]
```diff
@@ -77,7 +77,7 @@
 			type="textarea"
 			label="COM_MEDIA_FIELD_LEGAL_IMAGE_EXTENSIONS_LABEL"
 			description="COM_MEDIA_FIELD_LEGAL_IMAGE_EXTENSIONS_DESC"
-			default="bmp,gif,jpg,jpeg,png,webp"
+			default="bmp,gif,jpg,jpeg,png,webp,avif"
```
Comment (Quy) [inline administrator/components/com_media/config.xml:80]:
```suggestion
			default="avif,bmp,gif,jpg,jpeg,png,webp"
```

**Comment by brianteeman** [comment]
To test to see if your server supports avif see https://php.watch/versions/8.1/gd-avif

**Comment by wilsonge** [comment]
Also to check if you use patchtester you will need to manually update the parameters in media manager

**Comment by HLeithner** [comment]
thx

**Pull request merged**